### PR TITLE
Add ttnn.emptyOp

### LINKED ISSUE #373: [TTNN Dialect] Add ttnn.emptyOp
Add `ttnn.emptyOp` in TTNN dialect to match TTNN lib API.

### PR DESCRIPTION
**Pull request by svuckovicTT**
- Add `ttnn.emptyOp` to match TTNN lib API
- Update tests to expect `ttnn.empty()` instead of `ttnn.full()` when converting from `tensor.empty()`

Resolves #373

### CODE REVIEW

**Review by nsmithtt — APPROVED** [review]

File: runtime/lib/ttnn/program.cpp [hunk 1]
```diff
@@ -174,6 +177,22 @@ run(::tt::target::ttnn::ToMemoryConfigOp const *op, ::ttnn::Device &device,
...
+      op->out()->desc()->layout()->memory_desc()->data_type());
+  // TODO: determine layout, hardcoding tile_layout for now
+  auto desiredLayout = ::ttnn::Layout::TILE;
+  auto shape = ::ttnn::Shape(::tt::tt_metal::Shape(
+      utils::toShapeFromFBShape(*op->out()->desc()->shape())));
+  tensorPool.push_back(
+      ::ttnn::empty(shape, targetDataTypeTTNN, desiredLayout, device));
+  liveTensors.try_emplace(op->out()->global_id(), &tensorPool.back());
+}
+
```
Comment (svuckovicTT) [inline runtime/lib/ttnn/program.cpp:195]: @jnie-TT @tapspatel

Guys, would you mind checking this piece of code please?

I'd also like to know if you have any suggestions on handling the layout (tile vs row-major) here - I hardcoded it as I didn't find it in the structs of the output `TensorRef`.

File: runtime/lib/ttnn/program.cpp [hunk 2]
```diff
@@ -174,6 +177,22 @@ run(::tt::target::ttnn::ToMemoryConfigOp const *op, ::ttnn::Device &device,
   }
 }
 
+static void
+run(::tt::target::ttnn::EmptyOp const *op, ::ttnn::device::Device &device,
```
Comment (jnie-TT) [inline runtime/lib/ttnn/program.cpp:181]: Please use `::ttnn::Device` here

File: runtime/lib/ttnn/program.cpp (hunk 1 above)
Comment (jnie-TT) [reply runtime/lib/ttnn/program.cpp:195]: @svuckovicTT this is related to #272, the idea is once we emit the correct tile dims, we can determine the layout (1x1 untilized, 32x32 tilized)

Comment (svuckovicTT) [reply runtime/lib/ttnn/program.cpp:195]: Gotcha, so I'll leave it as-is for the time being.

**Review by jnie-TT — APPROVED** [review]

**Pull request merged**